File: app/src/main/java/com/setiawanpaiman/sunnyreader/util/GsonParser.java
package com.setiawanpaiman.sunnyreader.util;

import com.google.gson.Gson;

import java.lang.reflect.Type;

public class GsonParser implements JsonParser {

    private final Gson mGson;

    public GsonParser(Gson gson) {
        mGson = gson;
    }

    @Override
    public <T> T fromJson(String json, Type typeOfT) {
        return mGson.fromJson(json, typeOfT);
    }

    @Override
    public <T> String toJson(T src, Type typeOfSrc) {
        return mGson.toJson(src, typeOfSrc);
    }
}

File: app/src/main/java/com/setiawanpaiman/sunnyreader/util/GsonParser.kt
package com.setiawanpaiman.sunnyreader.util

import com.google.gson.Gson
import java.lang.reflect.Type

class GsonParser(private val mGson: Gson) : JsonParser {
    override fun <T> fromJson(json: String, typeOfT: Type): T {
        return mGson.fromJson(json, typeOfT)
    }

    override fun <T> toJson(src: T, typeOfSrc: Type): String {
        return mGson.toJson(src, typeOfSrc)
    }
}